### DISCUSSION TITLE
Question about running tests on chain forks in foundry fundamentals course

### DISCUSSION THREAD

**Discussion by accurec** [Cyfrin/foundry-full-course-cu#2985] [Q&A] [answered]
Hi! When we're running a command like

```
forge test --mt testPriceFeedVersionIsCorrect -vvvvv --fork-url ALCHEMY_URL_HERE
```

Why there is no account that we ned to provide and there is no need to spend any gas? Is it because foundry is smart enough to know that for example `new FundMe()` does not need to be deployed? I am assuming that we do not need to use spend any gas as well on the `return priceFeed.version()`, because it is a read from the chain call. What if we needed to write something to the chain - I am assuming that would not work with test suite from foundry?

**Comment by heheboii11** [comment, marked as answer]
Hey @accurec, as you can see you have used ```--fork-url```, which means you are forking the chain locally on your computer. So you are not interacting with the real blockchain, you are creating a fork locally and. running the tests there. So you dont need to provide any private keys. If you want to test it on real chain instead of using ```--fork-url``` just use ```--rpc-url```. Then you have to provide a private key and broadcast it to the chain. Hope this clears your doubt

**Reply by EngrPips** [reply]
Great heads-up.

**Reply by chainflash** [reply]
awesome 👍

**Reply by accurec** [reply]
@heheboii11 hi! Could you please explain in more detail what do you mean by forking the chain on my computer? I don't believe the whole sepolia chain somehow magically gets downloaded to my local using this process, so that's why I'm confused with your explanation a bit.

**Reply by accurec** [reply]
Okay from the docs:

```
-f, --fork-url <URL>
          Fetch state over a remote endpoint instead of starting from an empty state.
```

So I think it does not download the chain on my local but rather if there is a need to fetch a state, it will do it for that specific contract. So I would assume if there was a need to save a state in the test (for whatever reason), that would not work.

**Reply by EngrPips** [reply]
This makes sense.

**Reply by heheboii11** [reply]
> Okay from the docs:
> 
> ```
> -f, --fork-url <URL>
>           Fetch state over a remote endpoint instead of starting from an empty state.
> ```
> 
> So I think it does not download the chain on my local but rather if there is a need to fetch a state, it will do it for that specific contract. So I would assume if there was a need to save a state in the test (for whatever reason), that would not work.

Something like this but if you need more information you can go on to youtube or find some blog regarding forking a chain.

**Reply by EngrPips** [reply]
Indeed.
